chainer/training/extensions/micro_average.py

```diff
@@ -0,0 +1,72 @@
+import collections
+
+from chainer import reporter
+from chainer.training import extension
+from chainer.training import util
+
+
+class MicroAverage(extension.Extension):
```

Comment:
I feel this extension can be used for other metrics such as micro average precision/recall. But i also feel it is too general. How do you think?